src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -191,15 +191,62 @@ def remove_y_operators_from_circuit(self, start_index: int = 0) -> None:
 
             i += 1
 
-    def commute_pi_over_four_rotation(self, index: int) -> None:
-        """Commute a rotation block pass its neighbor block.
+    def __swap_adjacent_commuting_blocks(self, index: int) -> None:
+        """
+        Move a pi over four rotation block past its' neighbor block when the blocks commute
+
+        Args:
+            index (int): Index of the targeted block in the current circuit
+        """
+        next_block = index + 1
+        temp = self.ops[index]
+        self.ops[index] = self.ops[next_block]
+        self.ops[next_block] = temp
+
+    def __swap_adjacent_anticommuting_blocks(self, index: int) -> None:
+        """
+        Move a pi over four rotation block past its' neighbor block when the blocks anti-commute
 
         Args:
-            index (int): Index of the targeted block in the current circuit.
+            index (int): Index of the targeted block in the current circuit
 
         """
```

Comment:
And here an anti commuting assert

Reply:
The asserts are handled in the function `swap_adjacent_blocks()`. The respective private functions are only called if they commute or anti-commute

Reply:
Yes, they are now, but it's an invariant that needs to hold in the future. It's a strategic thing to protect code when adding new features and refactoring, because a future programmer might not be aware of this invariant (I know it's encoded in the name, but still...).

It just seem like a trivial change that can save so headaches to someone in the future

Reply:
I can add them not an issue! I did not this was a good coding practice :)